src/datasets/common.py

```diff
@@ -19,3 +26,70 @@ class SLDatasetSample(TypedDict):
 
     text: list[tuple[str, int]]
     label: int
+
+
+def collate_batch_common(slices: list[torch.Tensor], labels: list[int]):
+    # [batch size; most slices; font size; window size]
+    batched_slices = pad_sequence(slices, batch_first=True, padding_value=0.0).float()
+    bs, ms, _, _ = batched_slices.shape
+
+    # [batch size; most slices]
+    attention_mask = torch.zeros((bs, ms), dtype=torch.float)
+    for i, s in enumerate(slices):
+        attention_mask[i, : len(s)] = 1
+
+    return {
+        "slices": batched_slices,
+        "attention_mask": attention_mask,
+        "labels": torch.tensor(labels, dtype=torch.float),
+    }
+
+
+class AugmentationDataset(IterableDataset):
```

Comment:
Where is collate function here?
Since all our dataset classes contain their own collators, I think it would be easy in the future if this class also contains this method.